### PR TITLE
thinkfan: 0.9.3 -> 1.0.2 (replaces #55895)

### PR DESCRIPTION
**Pull request by cko**
###### Motivation for this change
#55895(thinkfan: 0.9.3 -> 1.0.1) wasn't merged due to some remarks and some merge conflicts with master. This PR incorporates the changes from #55895, addresses the remarks and adds an update to 1.0.2.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @domenkozar

### CODE REVIEW

File: pkgs/tools/system/thinkfan/default.nix [hunk 1]
```diff
@@ -1,18 +1,27 @@
-{ stdenv, fetchurl, cmake
+{ stdenv, fetchFromGitHub, cmakeCurses, libyamlcpp, pkgconfig
```
Comment (peterhoeg) [inline pkgs/tools/system/thinkfan/default.nix]: Why do you need curses support in cmake?

File: pkgs/tools/system/thinkfan/default.nix [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
-    sha256 = "0nz4c48f0i0dljpk5y33c188dnnwg8gz82s4grfl8l64jr4n675n";
+  src = fetchFromGitHub {
+    owner = "vmatare";
+    repo = "thinkfan";
+    rev = version;
+    sha256 = "107vw0962hrwva3wra9n3hxlbfzg82ldc10qssv3dspja88g8psr";
   };
 
-  nativeBuildInputs = [ cmake ];
+  configureFlags = [
```
Comment (peterhoeg) [inline pkgs/tools/system/thinkfan/default.nix]: ```configureFlags``` should be ```cmakeFlags```.

File: pkgs/tools/system/thinkfan/default.nix [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
+    "-DCMAKE_INSTALL_DOCDIR=share/doc/${pname}"
+    "-DUSE_NVML=OFF"
+    "-DUSE_ATASMART=ON"
+    "-DUSE_YAML=ON"
```
Comment (peterhoeg) [inline pkgs/tools/system/thinkfan/default.nix]: yaml is already on by default.

File: pkgs/tools/system/thinkfan/default.nix [hunk 4]
```diff
@@ -1,18 +1,27 @@
...
+  ];
 
-  buildInputs = stdenv.lib.optional smartSupport libatasmart;
+  nativeBuildInputs = [ cmakeCurses pkgconfig ];
+
+  buildInputs = [libyamlcpp] ++ stdenv.lib.optional smartSupport libatasmart;
```
Comment (peterhoeg) [inline pkgs/tools/system/thinkfan/default.nix]: lists have spaces before and after the first/last elements.

File: pkgs/tools/system/thinkfan/default.nix [hunk 5]
```diff
@@ -1,18 +1,27 @@
...
 
   cmakeFlags = stdenv.lib.optional smartSupport "-DUSE_ATASMART=ON";
```
Comment (peterhoeg) [inline pkgs/tools/system/thinkfan/default.nix]: This needs to move when you change configureflags above.

File: pkgs/tools/system/thinkfan/default.nix [hunk 6]
```diff
@@ -27,7 +36,7 @@ stdenv.mkDerivation rec {
 
   meta = {
```
Comment (peterhoeg) [inline pkgs/tools/system/thinkfan/default.nix]: if you do ```with stdenv.lib;``` here you can drop those as prefixes below.

File: pkgs/tools/system/thinkfan/default.nix [hunk 7]
```diff
@@ -27,7 +36,7 @@ stdenv.mkDerivation rec {
...
     license = stdenv.lib.licenses.gpl3;
-    homepage = http://thinkfan.sourceforge.net/;
+    homepage = https://github.com/vmatare/thinkfan;
```
Comment (peterhoeg) [inline pkgs/tools/system/thinkfan/default.nix]: Quote the url.

**Review by peterhoeg — CHANGES_REQUESTED** [review]

File: pkgs/tools/system/thinkfan/default.nix (hunk 1 above)
Comment (cko) [reply pkgs/tools/system/thinkfan/default.nix]: Thanks for your feedback! I pushed the requested changes.

File: pkgs/tools/system/thinkfan/default.nix [hunk 8]
```diff
@@ -25,10 +30,10 @@ stdenv.mkDerivation rec {
     install -Dm644 {src,$out/share/man/man1}/thinkfan.1
   '';
 
-  meta = {
-    license = stdenv.lib.licenses.gpl3;
-    homepage = http://thinkfan.sourceforge.net/;
-    maintainers = with stdenv.lib.maintainers; [ domenkozar ];
-    platforms = stdenv.lib.platforms.linux;
+  meta = with stdenv.lib; {
```
Comment (peterhoeg) [inline pkgs/tools/system/thinkfan/default.nix]: please also add a description - left that out earlier, sorry

Comment (cko) [reply pkgs/tools/system/thinkfan/default.nix]: done :-)

File: pkgs/tools/system/thinkfan/default.nix [hunk 9]
```diff
@@ -1,20 +1,25 @@
...
+    "-DUSE_NVML=OFF"
+  ] ++ stdenv.lib.optional smartSupport "-DUSE_ATASMART=ON";
 
-  buildInputs = stdenv.lib.optional smartSupport libatasmart;
+  nativeBuildInputs = [ cmake pkgconfig ];
 
-  cmakeFlags = stdenv.lib.optional smartSupport "-DUSE_ATASMART=ON";
+  buildInputs = [ libyamlcpp ] ++ stdenv.lib.optional smartSupport libatasmart;
 
   installPhase = ''
```
Comment (peterhoeg) [inline pkgs/tools/system/thinkfan/default.nix]: Last thing, sorry, please do ```runHook preInstall/postInstall``` when you are overriding the whole phase.

**Review by peterhoeg — APPROVED** [review]

**Pull request merged**